lambdas/cloudwatch_logs/main.js

```diff
@@ -0,0 +1,239 @@
+// Based on AWS Cloudwatch "Stream to Amazon Elasticsearch Service" funcion v1.1.2
+// Required stack parameters:
+// * EnvVarElasticsearchEndpoint - Amazon Elasticsearch Service domain endpoint
+
+var https = require('https');
+var zlib = require('zlib');
+var crypto = require('crypto');
+
+var environment = JSON.parse(require('fs').readFileSync('env', 'utf8'));
+var endpoint = environment['ELASTICSEARCH_ENDPOINT'];
+
+exports.handler = function(input, context) {
+    // decode input from base64
+    var zippedInput = new Buffer(input.awslogs.data, 'base64');
+
+    // decompress the input
+    zlib.gunzip(zippedInput, function(error, buffer) {
+        if (error) { context.fail(error); return; }
+
+        // parse the input from JSON
+        var awslogsData = JSON.parse(buffer.toString('utf8'));
+
+        // transform the input to Elasticsearch documents
+        var elasticsearchBulkData = transform(awslogsData);
+
+        // skip control messages
+        if (!elasticsearchBulkData) {
+            console.log('Received a control message');
+            context.succeed('Control message handled successfully');
+            return;
+        }
+
+        // post documents to the Amazon Elasticsearch Service
+        post(elasticsearchBulkData, function(error, success, statusCode, failedItems) {
+            console.log('Response: ' + JSON.stringify({
+                "statusCode": statusCode
+            }));
+
+            if (error) {
+                console.log('Error: ' + JSON.stringify(error, null, 2));
+
+                if (failedItems && failedItems.length > 0) {
+                    console.log("Failed Items: " +
+                        JSON.stringify(failedItems, null, 2));
+                }
+
+                context.fail(JSON.stringify(error));
+            } else {
+                console.log('Success: ' + JSON.stringify(success));
+                context.succeed('Success');
+            }
+        });
+    });
+};
+
+function transform(payload) {
+    if (payload.messageType === 'CONTROL_MESSAGE') {
+        return null;
+    }
+
+    var bulkRequestBody = '';
+
+    payload.logEvents.forEach(function(logEvent) {
+        var timestamp = new Date(1 * logEvent.timestamp);
+
+        // index name format: cwl-YYYY.MM.DD
+        var indexName = [
+            'cwl-' + timestamp.getUTCFullYear(),              // year
+            ('0' + (timestamp.getUTCMonth() + 1)).slice(-2),  // month
```

Comment:
Might be worth having a comment in the code about why we need `.slice(-2)` here as it seems a pretty funky thing to do. (I guess it is because of this? http://stackoverflow.com/questions/3605214/javascript-add-leading-zeroes-to-date)